ci/cscs.yml

```diff
@@ -0,0 +1,45 @@
+include:
+  - remote: 'https://gitlab.com/cscs-ci/recipes/-/raw/master/templates/v2/.ci-ext.yml'
+
+stages:
+  - prepare
+  - configure
+  - run
+
+variables:
+  CI_GENERATOR_IMAGE_NAME: $CSCS_REGISTRY_PATH/public/ci-generator:$CI_GENERATOR_TAG
+
+build-ci-generator-image:
+  stage: prepare
+  extends: .container-builder-cscs-zen2
+  before_script:
+    - echo $CI_GENERATOR_IMAGE_NAME
+    - export CI_GENERATOR_TAG=`cat ci/docker/* | sha256sum - | head -c 16`
+    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/public/ci-generator:$CI_GENERATOR_TAG
+    - echo "CI_GENERATOR_TAG=$CI_GENERATOR_TAG" > build.env
+  artifacts:
+    reports:
+      dotenv: build.env
+  variables:
+    DOCKERFILE: ci/docker/Dockerfile
+    PERSIST_IMAGE_NAME: $CI_GENERATOR_IMAGE_NAME
```

Comment:
`PERSIST_IMAGE_NAME: $CI_GENERATOR_IMAGE_NAME` is useless, it's overwritten in the `before_script` anyway.